platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/ReadableRosterStoreImpl.java

```diff
@@ -114,22 +114,16 @@ public Bytes getActiveRosterHash() {
         return latestRoundRosterPair.activeRosterHash();
     }
 
-    /** {@inheritDoc} */
     @Nullable
     @Override
-    public List<RoundRosterPair> getRosterHistory() {
-        final RosterState rosterStateSingleton = rosterState.get();
-        if (rosterStateSingleton != null
-                && !rosterStateSingleton.roundRosterPairs().isEmpty()) {
-            boolean allHashesPresent = rosterStateSingleton.roundRosterPairs().stream()
-                    .allMatch(pair -> rosterMap.get(ProtoBytes.newBuilder()
-                                    .value(pair.activeRosterHash())
-                                    .build())
-                            != null);
```

Comment:
While it's unclear if this validation logic belongs to the readable store, it's still important to have this implemented somewhere. I suggest to keep it for now. If you want to remove it from this method, then the logic should be moved elsewhere as a part of this same fix to ensure we maintain these invariants.

Reply:
Makes sense, added filtering to only return `RoundRosterPair`s that reference extant rosters.